OpenSim/Simulation/Wrap/WrapEllipsoid.cpp

```diff
@@ -72,62 +69,37 @@ WrapEllipsoid::~WrapEllipsoid()
 {
 }
 
-//_____________________________________________________________________________
-/**
-* Copy constructor.
-*
-* @param aWrapEllipsoid WrapEllipsoid to be copied.
-*/
-WrapEllipsoid::WrapEllipsoid(const WrapEllipsoid& aWrapEllipsoid) :
-   WrapObject(aWrapEllipsoid),
-   _dimensions(_dimensionsProp.getValueDblArray())
-{
-    setNull();
-    setupProperties();
-    copyData(aWrapEllipsoid);
-}
-
 //=============================================================================
 // CONSTRUCTION METHODS
 //=============================================================================
-//_____________________________________________________________________________
-/**
-* Set the data members of this WrapEllipsoid to their null values.
-*/
-void WrapEllipsoid::setNull()
-{
-}
-
 //_____________________________________________________________________________
 /**
 * Connect properties to local pointers.
 */
-void WrapEllipsoid::setupProperties()
+void WrapEllipsoid::constructProperties()
 {
     // BASE CLASS
     //WrapObject::setupProperties();
 
-    const double defaultDimensions[] = {-1.0, -1.0, -1.0};
-    _dimensionsProp.setName("dimensions");
-    _dimensionsProp.setValue(3, defaultDimensions);
-    _propertySet.append(&_dimensionsProp);
+    SimTK::Vec3 defaultDimensions = {1.0, 1.0, 1.0};
+    constructProperty_dimensions(defaultDimensions);
 }
 
 //_____________________________________________________________________________
 /**
 * Perform some set up functions that happen after the
 * object has been deserialized or copied.
 *
-* @param aModel 
+* @param aModel
 */
-void WrapEllipsoid::connectToModelAndBody(Model& aModel, PhysicalFrame& aBody)
+void WrapEllipsoid::extendFinalizeFromProperties()
 {
     // Base class
-    WrapObject::connectToModelAndBody(aModel, aBody);
+    WrapObject::extendFinalizeFromProperties();
 
     // maybe set a parent pointer, _body = aBody;
 
-    if (_dimensions[0] < 0.0 || _dimensions[1] < 0.0 || _dimensions[2] < 0.0)
+    if (get_dimensions()[0] < 0.0 || get_dimensions()[1] < 0.0 || get_dimensions()[2] < 0.0)
     {
         string errorMessage = "Error: dimensions for WrapEllipsoid " + getName() + " were either not specified, or are negative.";
```

Comment:
This error message is no longer correct: unspecified dimensions are positive and do not cause this conditional to be true.